### ISSUE TITLE
Exit Ticket

### ISSUE THREAD

**Issue by rohitd3** [rohitd3/manyFacesML#7]

**Comment by rohitd3** [comment]
<details><summary>Details</summary>
<p>

Link to the [video](https://www.youtube.com/watch?v=4hdVfROdYWY)

</p>
</details>

**Comment by peacekeeper6** [comment]
Tutorials and Guides Watched ~ 4 hours each
- https://www.kaggle.com/code/salmaneunus/image-classification-in-pytorch-cifar10
- https://www.kaggle.com/datasets/veeralakrishna/butterfly-dataset
- https://www.kaggle.com/datasets/arnaud58/flickrfaceshq-dataset-ffhq
- https://www.kaggle.com/datasets/andrewmvd/animal-faces
- https://github.com/jayrodge/Binary-Image-Classifier-PyTorch
- https://github.com/jayrodge/Binary-Image-Classifier-PyTorch/blob/master/Binary_face_classifier.ipynb 
- https://hackernoon.com/binary-face-classifier-using-pytorch-2d835ccb7816
- https://github.com/jayrodge/Binary-Image-Classifier-PyTorch/blob/master/Binary_face_classifier.ipynb
- https://stackoverflow.com/questions/52468956/how-do-i-visualize-a-net-in-pytorch
- https://www.youtube.com/playlist?list=PLqnslRFeH2UrcDBWF5mfPGpqQDSta6VK4 ~ 8 videos watched
  - Installation
  - Tensor Basics
  - Model, Loss, Optimizer
  - Linear Regression
  - Dataset and Dataloader - Batch
  - Dataset Transforms
  - Convulutional Neural Network(CNN)
  - Saving and Loading Models

| Names | Hours | Commits |
| -------- | -------- | -------- | 
| Everitt   |  ~17  |  [Commits](https://github.com/rohitd3/manyFacesML/commits?author=NinjaBreadLord) - Added blogposts explaining what PyTorch is, explained in between code, wrote code for displaying imagery and importing assets, dataset management, scrum management, markdown management, established dataset directory in google colab, installation of PyTorch, and bug/error handling in team code and own system. | 
| Rohit   |  17  |  [Commits](https://github.com/rohitd3/manyFacesML/commits?author=rohitd3) - Found helpful links (blogs/videos) that group could use to research. Utilized kaggle to find large datasets that could be used to train/validify/test CNN model. Explained the use of transformations to Samuel and Nathan. Made graph that details overfitting or underfitting in the model and also explained what could be addressed to fix those. Added calulating loss in the training and valid data. Graphed epoch results.   | 
| Nathan   |  17  |  [Commits](https://github.com/rohitd3/manyFacesML/commits?author=NathanShih04) - Worked with showing images and the transforms of the images. Researched and used plt or matplotlib.pyplot in order to show a visual demonstration of the transforms and data that we used. Worked on applying rotations and blurs to data and seperated input images into testing and training data. | 
| Jun   |    | [Commits](https://github.com/rohitd3/manyFacesML/commits?author=peacekeeper6)  | 
| Samuel   |  16  |  [Commits](https://github.com/rohitd3/manyFacesML/commits?author=Samuelwaang) - Helped Rohit make the graphs for loss data, explained the graphs, code for training neural network, a bunch of troubleshooting and error fixing for usingpytorch notebook file (imagesize, batch size, range, training, label, class size), explained training neural network code  |

**Comment by rohitd3** [comment]
Rohit Reflection:

Over the course of my journey through the AP CSA curriculum, it was a challenging journey  in the realm of computer science. Along the way, I encountered moments that heavily tested me. While certain concepts appeared difficult, the invaluable knowledge I acquired has strengthened  my understanding of the fundamental principles of computer science, particularly within the context of Java—an object-oriented programming language.
Through my encounters with Java, I have been mind blown by topics such as polymorphism and inheritance. The utilization of ArrayLists, Queue, and a multitude of sorting algorithms has allowed me to manipulate data. These newfound skills have not only enhanced my problem-solving abilities but have also sharpened my analytical thinking, equipping me with the tools necessary to tackle complex computational challenges.
Looking ahead, I am filled with anticipation for the next chapter of my academic journey as I aspire to pursue Computer Science and Engineering at UC Irvine. With the solid foundation laid by my experiences in both Computer Science Principles and Computer Science A, I hope to thrive in this interdisciplinary field. With my understanding of computer science concepts, I hope to contribute to the advancing technology and make a lasting impact on the world around me.

**Comment by Samuelwaang** [comment]
Samuel Reflection:

Throughout my time in CSA, I built upon my foundations in CSP of what it meant to be a coder. I was really able to start understanding code for what it did this year compared to last year. This year I feel like I felt like I understood or was able to start to understand things a lot better. 
One skill that I improved on this year that helped me a lot was self learning. Self learning was vital in filling holes of knowledge that I had. No matter how many lectures I would listen to, it wouldn't matter if I couldn't try and self learn. This would be through means of research or simply trial and error in code. Another skill I got better at throughout they year was teamwork. Especially first trimester, I was barely a team player. However, 2nd and 3rd tri, I learned a lot better to get help from my team members and also work with them. Code-wise, I learned a lot of java syntax and how to make data structures. I also started to really understand how databases and apis worked. Due the foundations I've built from this class, I'm confident that learning what I need in computer science will be a lot more manageable. I think it will also help my problem solving abilities in coding, as I will be able to think about more ideas of how to solve code issues and problems. 
I think I especially had a low with teamwork in first tri. I was especially quiet and didn't want to work with anyone. This led me to struggle and not learn as much as I should've that trimester. However, I've gotten into groups since that I've really enjoyed working in and I feel like my improved work in those trimesters reflects that. 
Now, I've committed to going to Ohio State University for computer science. I hope to challenge myself and will try to do internships and personal projects to enhance my computer science skills. This class has been a very good foundation for that. I hope to either go into a field of software engineering for any company or using computer science in the finance field. Hopefully my continued work into computer science will allow me to achieve this.
